core/sr-io/without_std.rs

```diff
@@ -331,8 +331,116 @@ pub mod ext {
 		// Offchain-worker Context
 		//================================
 
-		/// Submit extrinsic.
-		fn ext_submit_extrinsic(data: *const u8, len: u32);
+		/// Submit transaction.
+		///
+		/// Returns 0 if it was successfuly added to the pool, nonzero otherwise.
+		fn ext_submit_transaction(data: *const u8, len: u32) -> u32;
+
+		/// Sign a piece of data with authority key.
+		///
+		/// `sig_data` has to be a pointer to a slice of 64 bytes.
+		/// Returns `0` if successful, nonzero otherwise.
+		fn ext_sign(data: *const u8, len: u32, sig_data: *mut u8) -> u32;
+
+		/// Returns current UNIX timestamp (milliseconds)
+		fn ext_timestamp() -> u64;
+
+		/// Pause execution until given timestamp (milliseconds; `deadline`) is reached.
+		///
+		/// The deadline is obtained by querying the current timestamp via `ext_timestamp`
+		/// and then adding some time to it.
+		fn ext_sleep_until(deadline: u64);
+
+		/// Generate a random seed
+		///
+		/// `data` has to be a pointer to a slice of 32 bytes.
+		fn ext_random_seed(data: *mut u8);
+
+		/// Write a value to local storage.
+		fn ext_local_storage_set(key: *const u8, key_len: u32, value: *const u8, value_len: u32);
+
+		/// Read a value from local storage.
+		///
+		/// if `value_len` is u32::max_value the value has not been found.
```

Comment:
I'd like to see what happens with the result value in this case. If the return value is `NULL` in this case then can the user of this code rely this fact instead of checking for `max_value`? If not, why?   This is still unallocated memory then (or is the user expected to call `free` on this explicitly)?

Reply:
> So since the docs don't mention anything you can't rely on returned value to be NULL. I can add a clarification comment to all methods that have the same assumptions if you feel it's needed.

That would be great!